keras/src/saving/saving_api.py

```diff
@@ -15,6 +15,150 @@
+            # If compilation fails, leave the model uncompiled
+            pass
+
+    # Set the state in the model, but only for components that exist
+    state_to_set = {}
+
+    # Always load trainable and non-trainable variables
+    if "trainable_variables" in loaded_state:
+        state_to_set["trainable_variables"] = loaded_state[
+            "trainable_variables"
+        ]
+    if "non_trainable_variables" in loaded_state:
+        state_to_set["non_trainable_variables"] = loaded_state[
+            "non_trainable_variables"
+        ]
+
+    # Only load optimizer state if the model has an optimizer
+    if (
+        "optimizer_variables" in loaded_state
+        and hasattr(model, "optimizer")
+        and model.optimizer is not None
+    ):
+        # Ensure optimizer variables are created by doing a dummy
+        # apply_gradients. This creates the momentum/velocity
+        # variables that are needed
+        import numpy as np
+
+        # Create zero gradients for all trainable variables
+        zero_grads = [
+            backend.convert_to_tensor(np.zeros_like(v.numpy()))
+            for v in model.trainable_variables
+        ]
+        # Apply gradients to create optimizer slots
+        model.optimizer.apply_gradients(
+            zip(zero_grads, model.trainable_variables)
+        )
+        state_to_set["optimizer_variables"] = loaded_state[
+            "optimizer_variables"
+        ]
+
+    # Only load metrics state if the model has metrics variables
+    if (
+        "metrics_variables" in loaded_state
+        and hasattr(model, "metrics_variables")
+        and model.metrics_variables
+    ):
+        state_to_set["metrics_variables"] = loaded_state["metrics_variables"]
+
+    model.set_state_tree(state_to_set)
+
+    return model
+
+
+def _is_orbax_checkpoint(filepath):
+    """Check if the given path is an Orbax checkpoint directory."""
+    if not os.path.exists(filepath):
+        return False
+
+    try:
+        from keras.src.utils.module_utils import ocp
+
+        return ocp.is_orbax_checkpoint(filepath)
+    except (ImportError, AttributeError):
+        # Fallback to check for orbax.checkpoint file if Orbax API not available
+        return os.path.isfile(os.path.join(filepath, "orbax.checkpoint"))
+
+
+def _find_latest_orbax_checkpoint(checkpoint_dir):
+    """Find the latest checkpoint in an Orbax checkpoint directory."""
+    try:
+        from keras.src.utils.module_utils import ocp
+
+        checkpointer = ocp.training.Checkpointer(directory=checkpoint_dir)
+        latest = checkpointer.latest
+        if latest is None:
+            raise ValueError(f"No valid checkpoints found in {checkpoint_dir}")
+        return os.path.join(checkpoint_dir, str(latest.step))
+    except (ImportError, AttributeError):
+        raise ImportError(
+            "Orbax checkpoint support requires the 'orbax-checkpoint' package. "
+            "Install it with: pip install orbax-checkpoint"
+        )
```

Comment:
If you remove this try / except, you'll get the correct message from the LazyModule, no need to replicate this logic.

Reply:
Done